src/runtime_src/core/common/api/xrt_device.cpp

```diff
@@ -396,6 +396,87 @@ open_context(xrt::aie::device::access_mode am)
   core_device->open_aie_context(am);
 }
 
+static void
+get_abs_col(const xrt_core::device* device, uint16_t context_id, uint16_t& col)
```

Comment:
For the purpose of ordering, let's move this function before the member function definitions within this namespace, e.g. move to line 383.

This function should return the result.  It is confusing without comments that `col` is both used as input and adjusted.  The function should probably return the new adjusted value rather than modify a reference argument; so pass by value, adjust, return adjusted, and maybe comment to explain what it means to get abs col.